providers/qedr/qelr_verbs.c

```diff
@@ -922,14 +922,17 @@ static inline void qelr_edpm_set_inv_imm(struct qelr_qp *qp,
 					 struct qelr_edpm *edpm,
 					 uint32_t inv_key_or_imm_data)
 {
+	uint32_t data;
+
 	if (!edpm->is_edpm)
 		return;
 
-	memcpy(&edpm->dpm_payload[edpm->dpm_payload_offset],
-	       &inv_key_or_imm_data, sizeof(inv_key_or_imm_data));
+	data = htobe32(inv_key_or_imm_data);
+
```

Comment:
BTW, regarding 'inv_key_or_imm_data' , the verbs spec is that wc->imm_data is BE, and a NIC must achieve a certain order on the wire for compat.

However, when imm_data is being used as the invalidate_rkey (eg in SEND_INVALIDATE) then it is in host endian. 

I have a patch that adds a transparent union to make this sparse compatible.